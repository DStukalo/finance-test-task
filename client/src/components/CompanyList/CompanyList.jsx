import useTrickersFacade from '../../hooks/useStoreFacade';
import { Company } from './Company/Company';
import styles from './CompanyList.module.scss';

export function CompanyList() {
	const {
		loaded, AAPL, GOOGL, MSFT, AMZN, FB, TSLA,
	} = useTrickersFacade();

	return (
		<div className={styles.companies}>
			<h3 className={styles.companies__title}>
				Our Company
			</h3>
			{loaded ? (
				<div className={styles.list}>
					{AAPL ? (
						<Company
							fullname="Apple"
							name={AAPL.ticker}
							price={AAPL.price}
							percent={AAPL.change_percent}
							change={AAPL.change}
						/>
					) : null}
					{GOOGL ? (
						<Company
							fullname="Alphabet"
							name={GOOGL.ticker}
							price={GOOGL.price}
							percent={GOOGL.change_percent}
							change={GOOGL.change}
						/>
					) : null}
					{MSFT ? (
						<Company
							fullname="Microsoft"
							name={MSFT.ticker}
							price={MSFT.price}
							percent={MSFT.change_percent}
							change={MSFT.change}
						/>
					) : null}
					{AMZN ? (
						<Company
							fullname="Amazon"
							name={AMZN.ticker}
							price={AMZN.price}
							percent={AMZN.change_percent}
							change={AMZN.change}
						/>
					) : null}
					{FB ? (
						<Company
							fullname="Facebook"
							name={FB.ticker}
							price={FB.price}
							percent={FB.change_percent}
							change={FB.change}
						/>
					) : null}
					{TSLA ? (
						<Company
							fullname="Tesla"
							name={TSLA.ticker}
							price={TSLA.price}
							percent={TSLA.change_percent}
							change={TSLA.change}
						/>
					) : null}
				</div>
			) : (
				<div className={styles.loading}>loading...</div>
			)}
		</div>
	);
}
