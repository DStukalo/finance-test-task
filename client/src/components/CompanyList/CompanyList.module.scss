.companies {
	padding-bottom: 25px;
	display: flex;
	flex-direction: column;
	gap: 5px;
	width: 500px;

	@media (max-width: 899.99px) {
		margin: 0 auto;
	}

	@media (max-width: 520.99px) {
		width: max(95%, 300px);
	}

	&__title {
		font-size: var(--font-size-a1);
		padding: 5px;
	}
}
