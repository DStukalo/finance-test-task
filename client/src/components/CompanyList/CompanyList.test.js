import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { CompanyList } from './CompanyList';
import useTrickersFacade from '../../hooks/useStoreFacade';

jest.mock('../../hooks/useStoreFacade', () => jest.fn());

describe('CompanyList component', () => {
	it('CompanyList render with falsy values of store', () => {
		useTrickersFacade.mockImplementationOnce(() => ({
			loaded: false,
			AAPL: false,
			GOOGL: false,
			MSFT: false,
			AMZN: false,
			FB: false,
			TSLA: false,
		}));

		render(<CompanyList />, { wrapper: BrowserRouter });

		expect(screen.getByText('Our Company')).toHaveClass('companies__title');
		expect(screen.getByText('loading...')).toHaveClass('loading');
	});

	it('CompanyList render with correct values of store', () => {
		useTrickersFacade.mockImplementationOnce(() => ({
			loaded: true,
			AAPL: {
				change: '171.57',
				change_percent:	'0.35',
				dividend:	'0.10',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'137.33',
				ticker:	'AAPL',
				yield:	'0.76',
			},
			GOOGL: {
				change:	'175.21',
				change_percent:	'0.81',
				dividend:	'0.70',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'251.27',
				ticker:	'GOOGL',
				yield:	'0.67',
			},
			MSFT: {
				change:	'189.53',
				change_percent:	'0.46',
				dividend:	'0.34',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'141.77',
				ticker:	'MSFT',
				yield:	'1.33',
			},
			AMZN: {
				change:	'46.43',
				change_percent:	'0.46',
				dividend:	'0.68',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'205.48',
				ticker:	'AMZN',
				yield:	'0.01',
			},
			FB: {
				change:	'171.42',
				change_percent:	'0.09',
				dividend:	'0.95',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'241.02',
				ticker:	'FB',
				yield:	'1.49',
			},
			TSLA: {
				change:	'157.71',
				change_percent:	'0.67',
				dividend:	'0.98',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'296.22',
				ticker:	'TSLA',
				yield:	'1.59',
			},
		}));

		render(<CompanyList />, { wrapper: BrowserRouter });

		expect(screen.getByText('Our Company')).toHaveClass('companies__title');
		expect(screen.getByText('Apple')).toHaveClass('company__name');
		expect(screen.getByText('Alphabet')).toHaveClass('company__name');
		expect(screen.getByText('Microsoft')).toHaveClass('company__name');
		expect(screen.getByText('Amazon')).toHaveClass('company__name');
		expect(screen.getByText('Facebook')).toHaveClass('company__name');
		expect(screen.getByText('Tesla')).toHaveClass('company__name');
	});

	it('CompanyList snapshot with falsy values of store', () => {
		useTrickersFacade.mockImplementationOnce(() => ({
			loaded: false,
			AAPL: false,
			GOOGL: false,
			MSFT: false,
			AMZN: false,
			FB: false,
			TSLA: false,
		}));

		const companyList = render(<CompanyList />, { wrapper: BrowserRouter });

		expect(companyList).toMatchSnapshot();
	});

	it('CompanyList snapshot with correct values of store', () => {
		useTrickersFacade.mockImplementationOnce(() => ({
			loaded: true,
			AAPL: {
				change: '171.57',
				change_percent:	'0.35',
				dividend:	'0.10',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'137.33',
				ticker:	'AAPL',
				yield:	'0.76',
			},
			GOOGL: {
				change:	'175.21',
				change_percent:	'0.81',
				dividend:	'0.70',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'251.27',
				ticker:	'GOOGL',
				yield:	'0.67',
			},
			MSFT: {
				change:	'189.53',
				change_percent:	'0.46',
				dividend:	'0.34',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'141.77',
				ticker:	'MSFT',
				yield:	'1.33',
			},
			AMZN: {
				change:	'46.43',
				change_percent:	'0.46',
				dividend:	'0.68',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'205.48',
				ticker:	'AMZN',
				yield:	'0.01',
			},
			FB: {
				change:	'171.42',
				change_percent:	'0.09',
				dividend:	'0.95',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'241.02',
				ticker:	'FB',
				yield:	'1.49',
			},
			TSLA: {
				change:	'157.71',
				change_percent:	'0.67',
				dividend:	'0.98',
				exchange:	'NASDAQ',
				last_trade_time:	'2023-04-11T10:49:05.000Z',
				price:	'296.22',
				ticker:	'TSLA',
				yield:	'1.59',
			},
		}));

		const companyList = render(<CompanyList />, { wrapper: BrowserRouter });

		expect(companyList).toMatchSnapshot();
	});
});
