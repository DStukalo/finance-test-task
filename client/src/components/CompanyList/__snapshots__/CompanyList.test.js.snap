// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyList component CompanyList snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="companies"
      >
        <h3
          class="companies__title"
        >
          Our Company
        </h3>
        <div
          class="loading"
        >
          loading...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="companies"
    >
      <h3
        class="companies__title"
      >
        Our Company
      </h3>
      <div
        class="loading"
      >
        loading...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CompanyList component CompanyList snapshot with correct values of store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="companies"
      >
        <h3
          class="companies__title"
        >
          Our Company
        </h3>
        <div
          class="list"
        >
          <div
            class="company"
          >
            <div
              class="company__brand"
            >
              <div
                class="Apple"
              />
              <div
                class="company__name"
              >
                Apple
              </div>
            </div>
            <p
              class="company__price"
            >
              137.33
              $
            </p>
            <div
              class="company__down"
            >
              <p>
                0.35
                %
              </p>
              <div
                class="company__down_logo"
              />
            </div>
            <div
              class="company__change_down"
            >
              -
              171.57
              $
            </div>
            <button
              class="company__btn"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="company"
          >
            <div
              class="company__brand"
            >
              <div
                class="Alphabet"
              />
              <div
                class="company__name"
              >
                Alphabet
              </div>
            </div>
            <p
              class="company__price"
            >
              251.27
              $
            </p>
            <div
              class="company__down"
            >
              <p>
                0.81
                %
              </p>
              <div
                class="company__down_logo"
              />
            </div>
            <div
              class="company__change_down"
            >
              -
              175.21
              $
            </div>
            <button
              class="company__btn"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="company"
          >
            <div
              class="company__brand"
            >
              <div
                class="Microsoft"
              />
              <div
                class="company__name"
              >
                Microsoft
              </div>
            </div>
            <p
              class="company__price"
            >
              141.77
              $
            </p>
            <div
              class="company__down"
            >
              <p>
                0.46
                %
              </p>
              <div
                class="company__down_logo"
              />
            </div>
            <div
              class="company__change_down"
            >
              -
              189.53
              $
            </div>
            <button
              class="company__btn"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="company"
          >
            <div
              class="company__brand"
            >
              <div
                class="Amazon"
              />
              <div
                class="company__name"
              >
                Amazon
              </div>
            </div>
            <p
              class="company__price"
            >
              205.48
              $
            </p>
            <div
              class="company__down"
            >
              <p>
                0.46
                %
              </p>
              <div
                class="company__down_logo"
              />
            </div>
            <div
              class="company__change_down"
            >
              -
              46.43
              $
            </div>
            <button
              class="company__btn"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="company"
          >
            <div
              class="company__brand"
            >
              <div
                class="Facebook"
              />
              <div
                class="company__name"
              >
                Facebook
              </div>
            </div>
            <p
              class="company__price"
            >
              241.02
              $
            </p>
            <div
              class="company__down"
            >
              <p>
                0.09
                %
              </p>
              <div
                class="company__down_logo"
              />
            </div>
            <div
              class="company__change_down"
            >
              -
              171.42
              $
            </div>
            <button
              class="company__btn"
              type="button"
            >
              +
            </button>
          </div>
          <div
            class="company"
          >
            <div
              class="company__brand"
            >
              <div
                class="Tesla"
              />
              <div
                class="company__name"
              >
                Tesla
              </div>
            </div>
            <p
              class="company__price"
            >
              296.22
              $
            </p>
            <div
              class="company__down"
            >
              <p>
                0.67
                %
              </p>
              <div
                class="company__down_logo"
              />
            </div>
            <div
              class="company__change_down"
            >
              -
              157.71
              $
            </div>
            <button
              class="company__btn"
              type="button"
            >
              +
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="companies"
    >
      <h3
        class="companies__title"
      >
        Our Company
      </h3>
      <div
        class="list"
      >
        <div
          class="company"
        >
          <div
            class="company__brand"
          >
            <div
              class="Apple"
            />
            <div
              class="company__name"
            >
              Apple
            </div>
          </div>
          <p
            class="company__price"
          >
            137.33
            $
          </p>
          <div
            class="company__down"
          >
            <p>
              0.35
              %
            </p>
            <div
              class="company__down_logo"
            />
          </div>
          <div
            class="company__change_down"
          >
            -
            171.57
            $
          </div>
          <button
            class="company__btn"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="company"
        >
          <div
            class="company__brand"
          >
            <div
              class="Alphabet"
            />
            <div
              class="company__name"
            >
              Alphabet
            </div>
          </div>
          <p
            class="company__price"
          >
            251.27
            $
          </p>
          <div
            class="company__down"
          >
            <p>
              0.81
              %
            </p>
            <div
              class="company__down_logo"
            />
          </div>
          <div
            class="company__change_down"
          >
            -
            175.21
            $
          </div>
          <button
            class="company__btn"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="company"
        >
          <div
            class="company__brand"
          >
            <div
              class="Microsoft"
            />
            <div
              class="company__name"
            >
              Microsoft
            </div>
          </div>
          <p
            class="company__price"
          >
            141.77
            $
          </p>
          <div
            class="company__down"
          >
            <p>
              0.46
              %
            </p>
            <div
              class="company__down_logo"
            />
          </div>
          <div
            class="company__change_down"
          >
            -
            189.53
            $
          </div>
          <button
            class="company__btn"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="company"
        >
          <div
            class="company__brand"
          >
            <div
              class="Amazon"
            />
            <div
              class="company__name"
            >
              Amazon
            </div>
          </div>
          <p
            class="company__price"
          >
            205.48
            $
          </p>
          <div
            class="company__down"
          >
            <p>
              0.46
              %
            </p>
            <div
              class="company__down_logo"
            />
          </div>
          <div
            class="company__change_down"
          >
            -
            46.43
            $
          </div>
          <button
            class="company__btn"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="company"
        >
          <div
            class="company__brand"
          >
            <div
              class="Facebook"
            />
            <div
              class="company__name"
            >
              Facebook
            </div>
          </div>
          <p
            class="company__price"
          >
            241.02
            $
          </p>
          <div
            class="company__down"
          >
            <p>
              0.09
              %
            </p>
            <div
              class="company__down_logo"
            />
          </div>
          <div
            class="company__change_down"
          >
            -
            171.42
            $
          </div>
          <button
            class="company__btn"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="company"
        >
          <div
            class="company__brand"
          >
            <div
              class="Tesla"
            />
            <div
              class="company__name"
            >
              Tesla
            </div>
          </div>
          <p
            class="company__price"
          >
            296.22
            $
          </p>
          <div
            class="company__down"
          >
            <p>
              0.67
              %
            </p>
            <div
              class="company__down_logo"
            />
          </div>
          <div
            class="company__change_down"
          >
            -
            157.71
            $
          </div>
          <button
            class="company__btn"
            type="button"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CompanyList component CompanyList snapshot with falsy values of store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="companies"
      >
        <h3
          class="companies__title"
        >
          Our Company
        </h3>
        <div
          class="loading"
        >
          loading...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="companies"
    >
      <h3
        class="companies__title"
      >
        Our Company
      </h3>
      <div
        class="loading"
      >
        loading...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
